test/pr.ts

```diff
@@ -23,60 +23,239 @@ before(() => {
 });
 
 describe('Opening a pull request', async () => {
+  const sandbox = sinon.createSandbox();
   const upstream = {owner: 'upstream-owner', repo: 'upstream-repo'};
   const origin = {
     owner: 'origin-owner',
     repo: 'origin-repo',
     branch: 'pr-test-branch',
   };
-  const responseData = await import('./fixtures/create-pr-response.json');
   const description = {
     title: 'PR-TITLE',
     body: 'PR-DESCRIPTION',
   };
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('Invokes octokit pull create when there is not an existing pull request open', async () => {
+    // setup
+    const responseCreatePullData = await import(
+      './fixtures/create-pr-response.json'
+    );
+    const createPrResponse = {
+      headers: {},
+      status: 200,
+      url: 'http://fake-url.com',
+      data: responseCreatePullData,
+    };
+    const listPullResponse = {
+      headers: {},
+      status: 200,
+      url: 'http://fake-url.com',
+      data: [],
+    };
+    sandbox.stub(octokit.pulls, 'list').resolves(listPullResponse);
+    const stub = sandbox
+      .stub(octokit.pulls, 'create')
+      .resolves(createPrResponse);
+    // tests
+    await openPullRequest(octokit, upstream, origin, description);
+    sandbox.assert.calledOnceWithExactly(stub, {
+      owner: upstream.owner,
+      repo: origin.repo,
+      title: description.title,
+      head: 'origin-owner:pr-test-branch',
+      base: 'master',
+      body: description.body,
+      maintainer_can_modify: true,
+    });
+  });
+
+  describe('When there are similar refs with pull requests open, the current new and unique ref still opens a pr', async () => {
+    const responseCreatePullData = await import(
+      './fixtures/create-pr-response.json'
+    );
+    const responseListPullData = await import(
+      './fixtures/list-pulls-response.json'
+    );
+    afterEach(() => {
+      sandbox.restore();
```

Comment:
This should be handled by the outer describe blocks's `afterEach` which is restoring the sandbox.